static/app/components/core/compactSelect/control.tsx

```diff
@@ -467,19 +440,22 @@ export function Control({
     },
   });
 
-  const showClearButton = useMemo(
-    () => selectedOptions.flat().length > 0,
-    [selectedOptions]
-  );
+  const hasSelection = useMemo(() => {
+    const selectedOptions = Array.isArray(value)
+      ? value
+      : value === undefined
+        ? []
+        : [value];
+    return selectedOptions.flat().length > 0;
```

Comment:
Not sure why we need the `array.flat()` check here?

```suggestion
  if (value === undefined) return false;
  if (Array.isArray(value)) return value.length > 0;
  return true;
```